sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -198,20 +203,26 @@ private static long getRequestContentLength(BinaryDataContent content, HttpHeade
     }
 
     private static HttpResponse toHttpResponse(HttpRequest request, okhttp3.Response response,
-        boolean eagerlyReadResponse, boolean eagerlyConvertHeaders) throws IOException {
+        boolean eagerlyReadResponse, boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
+        // Ignoring the response body takes precedent over eagerly reading the response body.
+        // Both should never be true at the same time but this is acts as a safeguard.
+        if (ignoreResponseBody) {
+            ResponseBody body = response.body();
+            if (body != null) {
+                body.close();
+            }
```

Comment:
When would `body` be non-null? If this check evaluates to true, then we should throw as we are ignoring response body when there's actually some content.

Reply:
Thoughts on making this a warning or info log message for now? This would be a possible issue where either the Swagger was wrong or the server unexpectantly returned a response. FWIW, we already ignore the response body when the return body type is void, this just pushes the logic down into the HTTP client layer.

Reply:
Done